src/modules/edit.js

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable prefer-const */
+/* eslint-disable import/no-mutable-exports */
+/* eslint-disable no-else-return */
+/* eslint-disable no-useless-return */
+/* eslint-disable array-callback-return */
+/* eslint-disable consistent-return */
+/* eslint-disable arrow-body-style */
+
+let x = JSON.parse(localStorage.getItem('savedData'));
+export let tasks = [];
+
+if (x) {
+  tasks = x;
+}
+
+export default function edit(e) {
+  const item = e.target;
+
+  if (item.classList.contains('fa-trash-can')) {
+    const removeParent = item.parentElement.parentElement;
+    removeParent.remove();
+
+    const removeTaskFromArray = item.parentElement.previousElementSibling.children[0].getAttribute('name');
+
+    tasks = tasks
+      .filter((newTask) => {
+        if (newTask.index !== +removeTaskFromArray) {
+          return true;
+        }
+      })
+      .sort((a, b) => {
+        if (a.index !== 1) {
+          a.index = 1;
+        }
+        return a.index - b.index;
+      })
+      .sort((task, newTask) => {
+        if (task.index + 1 === newTask.index) {
+          return;
+        } else {
+          newTask.index = task.index + 1;
+        }
```

Comment:
- [ ] If I create two tasks and then delete the first task, then the other index should be updated to become 1. But, in your case, it is still 2. It's working if it was more than 2 task remains but not if 1 task remains.

![Screenshot from 2022-04-29 00-11-07](https://user-images.githubusercontent.com/22434660/165824023-4dac2120-cede-4d76-8845-657712a2522c.png)